var/spack/repos/builtin/packages/py-scipy/package.py

```diff
@@ -206,29 +197,12 @@ def install(self, spec, prefix):
         if lapack == "armpl":
             lapack += "-dynamic-lp64-seq"
 
-        args = [
-            "setup",
-            "build",
-            "-Dblas=" + blas,
-            "-Dlapack=" + lapack,
-            "--prefix=" + join_path(os.getcwd(), "build-install"),
-            "-Ddebug=false",
-            "-Doptimization=2",
-        ]
-        meson = which("meson")
-        meson(*args)
-        args = [
-            "-m",
-            "build",
-            "--wheel",
-            "-Cbuilddir=build",
-            "--no-isolation",
-            "--skip-dependency-check",
-            ".",
-        ]
-        python(*args)
-        args = std_pip_args + ["--prefix=" + prefix, glob.glob(join_path("dist", "scipy*.whl"))[0]]
-        pip(*args)
+        return {
+            "blas": blas,
+            "lapack": lapack,
+            "debug": False,
+            "optimization": 2,
+        }
```

Comment:
https://meson-python.readthedocs.io/en/latest/reference/config-settings.html

meson-python expects config-settings for "setup-args", the value of which is a series of `-Dopt=val` arguments to Meson.

Reply:
So the flag would look something like:
```console
$ pip install --config-settings=setup-args=-Dblas=openblas
```
? Boy that's a mouthful.

Reply:
With `python -m build` it's a bit simpler because `--config-settings` becomes `-C`.

But yeah, kind of awkward.

Reply:
Okay, see if the new version looks better. Not sure whether multiple args would look like:
```console
$ pip install --config-settings=setup-args=-Dblas=openblas --config-settings=setup-args=-Dlapack=openblas
```
or
```console
$ pip install --config-settings=setup-args="-Dblas=openblas -Dlapack=openblas"
```

Reply:
The problem here is that `pip` still does not accept duplicate keys. We're in the process of fixing that - my edit to PEP 517 just got accepted. So you need to use `python -m build` to build a wheel, and then `pip install dist/scipy*.whl` to install it.

Reply:
Would my latter syntax avoid the duplicate key issue?

Reply:
No, there will be no string parsing/splitting.

Reply:
Alright, I'll just close this PR until pip properly supports meson.

Reply:
I also like the idea of build/installer, but based on #35326, we may be a ways off from using them in Spack instead of pip.